pkg/portal/cluster/machines.go

```diff
@@ -59,6 +68,59 @@ func (c *client) Machines(ctx context.Context) (*MachineListInformation, error)
 	return c.fetcher.Machines(ctx)
 }
 
+// VM Allocation Status Stuff
+type VMAllocationStatus map[string]string
+
+func (c *client) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	return c.fetcher.VMAllocationStatus(ctx)
+}
+
+func (f *realFetcher) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	env := f.azureSideFetcher.env
+	subscriptionDoc := f.azureSideFetcher.subscriptionDoc
+	clusterRGName := f.azureSideFetcher.resourceGroupName
+	fpAuth, err := env.FPAuthorizer(subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Getting Virtual Machine resources through the Cluster's Resource Group
+	computeResources, err := NewResourceClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth).ListByResourceGroup(ctx, clusterRGName, "resourceType eq 'Microsoft.Compute/virtualMachines'", "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	vmAllocationStatus := make(VMAllocationStatus)
+	virtualMachineClient := NewVirtualMachineClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth)
+	for _, res := range computeResources {
+		var vmName, allocationStatus string
+		if *res.Type != "Microsoft.Compute/virtualMachines" {
```

Comment:
Should we leave out this `if` statement since we're already using the `resourceType eq 'Microsoft.Compute/virtualMachines` filter in the call to `ListByResourceGroup` above?

Reply:
I think yes .. I am not able to recall .. why I put that condition there .. looking at the code now, it looks like the `if` condition is not required, hence removing it.